Issue/21189

STF-Zero opened the pull request:
Fixes #

-----

 ## Description:
This PR is related to [issue#21189](https://github.com/wordpress-mobile/WordPress-Android/issues/21189). 
This issue shows a problem that 'There's no visual confirmation when clicking UNDO button to restore the blog previously blocked'.
 Here's the issue video:

https://github.com/user-attachments/assets/f8b1dd31-a89d-4964-ad90-fe584117f7e5



## Solutions:
1.Add a toast message on the screen when clicking the UNDO button by snackbayMessage:
![image](https://github.com/user-attachments/assets/cfd31c24-7bc3-4f4b-b3cb-cd8c2d98f47d)
2.Add a new toast message into strings.xml:
![image](https://github.com/user-attachments/assets/dc1270e0-a38c-4fe4-ac70-cb29db683cb3)

## Result:

https://github.com/user-attachments/assets/04de8579-a02d-4b43-bd6f-1b5de08b1d2e



## To Test:

<!-- Test instructions per dependency update: https://github.com/wordpress-mobile/WordPress-Android/blob/trunk/docs/test_instructions_per_dependency_update.md -->

-----

## Regression Notes

1. Potential unintended areas of impact

    Simply add a toast message on the screen may not have impact on other areas which I have not confirmed yet. I'll keep looking for influence caused by my changes and work on to solve it if necessary!

3. What I did to test those areas of impact (or what existing automated tests I relied on)

    Still need to do.

4. What automated tests I added (or what prevented me from doing so)

    Still need to do.

-----

## PR Submission Checklist:

- [x] I have completed the Regression Notes.
- [x] I have considered adding accessibility improvements for my changes.
- [ ] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

-----

## Testing Checklist (strike-out the not-applying and unnecessary ones):

- [ ] WordPress.com sites and self-hosted Jetpack sites.
- [ ] Portrait and landscape orientations.
- [ ] Light and dark modes.
- [ ] Fonts: Larger, smaller and bold text.
- [ ] High contrast.
- [ ] Talkback.
- [ ] Languages with large words or with letters/accents not frequently used in English.
- [ ] Right-to-left languages. (Even if translation isn’t complete, formatting should still respect the right-to-left layout)
- [ ] Large and small screen sizes. (Tablet and smaller phones)
- [ ] Multi-tasking: Split screen and Pop-up view. (Android 10 or higher)

STF-Zero committed:
Update ReaderPostCardActionsHandler.kt

STF-Zero committed:
Update strings.xml

STF-Zero committed:
Merge branch 'trunk' into issue/21189

nbradbury commented on WordPress/src/main/res/values/strings.xml:
```diff
@@ -2300,6 +2300,7 @@
     <string name="reader_toast_err_unable_to_unfollow_blog">Unable to unsubscribe from blog</string>
     <string name="reader_toast_posts_from_this_blog_blocked">Posts from this blog will no longer be shown</string>
     <string name="reader_toast_err_unable_to_block_blog">Unable to block this blog</string>
+    <string name="reader_toast_success_UNDO">This blog has been unblocked!\nPosts from this blog will still be shown</string>
```
This is a minor nit, but the uppercase "UNDO" isn't standard. Can we stick with lowercase?

Also, let's shorten the message to just "Blog unblocked" without punctuation, which is more in line with the [Material writing guidelines](https://m2.material.io/design/communication/writing.html#principles).

STF-Zero replied:
Works for me! 
I'll change the string to lowercase and shorten the message to "Blog unblocked".
Thanks for the advice!

STF-Zero replied:
![c2c6ced610f353558e0d748b87da8c7](https://github.com/user-attachments/assets/237a7c13-bbce-4a69-92db-b03f0ae116e2)

STF-Zero committed:
Merge branch 'trunk' into issue/21189

STF-Zero committed:
Update strings.xml

change the string name to lowercase
(reader_toast_success_UNDO -> reader_toast_success_undo)
shorten the toast message.

STF-Zero committed:
Update ReaderPostCardActionsHandler.kt

nbradbury reviewed (approved):
@STF-Zero Thanks for working on this, changes look good! :shipit:
